pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -39,12 +43,15 @@ public Object processRequest(Request request, Response response, String userId)
             throw new RuntimeException("ColumnNameMap is null!");
         }
         String userIdRequest = UserUtil.getUserId(request);
-        if (UserUtil.checkUserAccess(null, userId, DBConstants.MR_VIEW, userIdRequest)
-                || UserUtil.checkUserAccess(null, userId, DBConstants.MR_ABSTRACTER, userIdRequest)
-                || UserUtil.checkUserAccess(null, userId, DBConstants.PT_LIST_VIEW, userIdRequest)) {
+        String requestBody = request.body();
```

Comment:
Looks like some exception throwing calls are no longer protected by try?

Reply:
I am not sure I understand...

Reply:
Let's say 'fromJson' throws an exception. In the past we would catch that and respond with a 500. Now it looks like we will not catch that? Is there something in Jetty/Spark that will catch that and convert it to a 500?
Or am I reading the code wrong?

Reply:
@denniscunningham - TIL there's a bit of "spark" magic that goes on but two other things to note. we have the ability to wire up exception handling when we wire up the route.  It doesn't look like we currently do that.  The default exception handling in spark's GeneralError implementation seems to protect against leaks when a CustomErrorPage is not set up.  It simply returns a 500 with the text: "<html><body><h2>500 Internal Server Error</h2></body></html>"

Reply:
Thank you both, this had skipped my eyes. I've now surrounded the whole method with try-catch.